src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -119,6 +119,11 @@ final class ListBuffer[A]
   // Don't use the inherited size, which forwards to a List and is O(n).
   override def size = length
 
+  // Override with efficient implementations using the extra size information available to ListBuffer.
+  override def isEmpty: Boolean = len == 0
+  override def nonEmpty: Boolean = len > 0
+  override def head: A = if (len > 0) start.asInstanceOf[::[A]].head else Nil.head
```

Comment:
This one doesn't look very cute.. `start.head` would be a virtual call with two cases, so C2 should be able to inline it. But before C2 it's probably slower than the monomorphic calls.